internal/report/create.go

```diff
@@ -0,0 +1,140 @@
+package report
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/kubewarden/audit-scanner/internal/constants"
+	policiesv1 "github.com/kubewarden/kubewarden-controller/pkg/apis/policies/v1"
+	"github.com/rs/zerolog/log"
+	admv1 "k8s.io/api/admission/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/wg-policy-prototypes/policy-report/pkg/api/wgpolicyk8s.io/v1beta1"
+)
+
+type PolicyReport v1beta1.PolicyReport
+type ClusterPolicyReport v1beta1.ClusterPolicyReport
+
+const PolicyReportSource string = "kubewarden"
+
+var clusterPolicyReport = ClusterPolicyReport{
+	ObjectMeta: metav1.ObjectMeta{Name: "polr-clusterwide"},
+	Summary: v1beta1.PolicyReportSummary{
+		Pass:  0, // count of policies with requirements met
+		Fail:  0, // count of policies with requirements not met
+		Warn:  0, // not used for now
+		Error: 0, // count of policies that couldn't be evaluated
+		Skip:  0, // count of policies that were not selected for evaluation
+	},
+	Results: []*v1beta1.PolicyReportResult{},
+}
+
+// slice of namespaced PolicyReports
+var policyReports = make(map[string]*PolicyReport)
```

Comment:
This could be a store struct with a mutex, accompanied with a policyReportStore interface with Add(), Update(), Get(), etc.

Reply:
If we return a "report object"  to the scanner, we reduce the need to use mutexes as well. As the scanner run sequentially, there is no need to add a mutex there as well. Furthermore, different report objects can be used in different threads (gorotines). Where each one will scanning different namespaces with not concurrent access (depending on how we implement it, of cource). But I think  this is a something we can do later.

Reply:
resolving this, now it is a proper store